### PR TITLE
Populate latency field in final_info.

### PR DESCRIPTION
**Pull request by markdroth**
This is the change requested by @bogdandrutu.

### CODE REVIEW

File: src/core/lib/surface/call.c [hunk 1]
```diff
@@ -427,6 +429,8 @@ static void destroy_call(grpc_exec_ctx *exec_ctx, void *call,
 
   get_final_status(call, set_status_value_directly,
                    &c->final_info.final_status);
+  c->final_info.stats.latency =
+      gpr_time_sub(gpr_now(GPR_CLOCK_MONOTONIC), c->start_time);
```
Comment (dgquintas) [inline src/core/lib/surface/call.c]: I like asserting. In part for documentation purposes, in part because I have a (n involuntary) thing for introducing bugs. Just a soft suggestion to add `GPR_ASSERT( gpr_time_cmp(c->final_info.stats.latency, gpr_time_0(GPR_CLOCK_MONOTONIC) > 0)` after the new code.

File: test/core/end2end/tests/filter_latency.c [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+  shutdown_server(f);
+  shutdown_client(f);
+
+  grpc_completion_queue_shutdown(f->cq);
+  drain_cq(f->cq);
+  grpc_completion_queue_destroy(f->cq);
+}
+
+// Simple request via a server filter that always fails to initialize
+// the call.
```
Comment (dgquintas) [inline test/core/end2end/tests/filter_latency.c]: copy-paste leftover? Seems unrelated/wrong.

File: test/core/end2end/tests/filter_latency.c [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+
+  grpc_byte_buffer_destroy(request_payload);
+  grpc_byte_buffer_destroy(request_payload_recv);
+
+  end_test(&f);
+  config.tear_down_data(&f);
+}
+
+/*******************************************************************************
+ * Test filter
```
Comment (dgquintas) [inline test/core/end2end/tests/filter_latency.c]: Test _latency_ filter?

File: test/core/end2end/tests/filter_latency.c [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+  grpc_call_destroy(c);
+  grpc_call_destroy(s);
+
+  const gpr_timespec end_time = gpr_now(GPR_CLOCK_MONOTONIC);
+  const gpr_timespec max_latency = gpr_time_sub(end_time, start_time);
+
+  GPR_ASSERT(gpr_time_cmp(max_latency, g_client_latency) >= 0);
+  GPR_ASSERT(gpr_time_cmp(gpr_time_0(GPR_TIMESPAN), g_client_latency) < 0);
+  GPR_ASSERT(gpr_time_cmp(max_latency, g_server_latency) >= 0);
+  GPR_ASSERT(gpr_time_cmp(gpr_time_0(GPR_TIMESPAN), g_server_latency) < 0);
```
Comment (dgquintas) [inline test/core/end2end/tests/filter_latency.c]: It'd be nice if we could test some relationship between the client and server latencies.

Also, if we added a sleep for a known amount of time, say `T`, at server we could assert that the `T < latency < T + epsilon`, for a reasonable `epsilon` making sure we are correctly measuring any work that goes on in the server, not just some random positive overhead if the measuring code were buggy.

File: src/core/lib/surface/call.c (hunk 1 above)
Comment (markdroth) [inline src/core/lib/surface/call.c]: I generally prefer to avoid assertions whenever possible, because it's always better for an application to return an error than to crash.  And in this particular case, the assertion wouldn't even add any benefit, because nothing would break if it's not true -- we'd simply get a bogus latency value returned.

File: test/core/end2end/tests/filter_latency.c (hunk 2 above)
Comment (markdroth) [inline test/core/end2end/tests/filter_latency.c]: Yeah.  I started this file by copying filter_call_init_fails.c, and I obviously didn't notice that this needed to be updated.  Fixed.

File: test/core/end2end/tests/filter_latency.c (hunk 4 above)
Comment (markdroth) [inline test/core/end2end/tests/filter_latency.c]: No need for a sleep here.  The client call is always created before the server call, so I can destroy the server call before the client call and then verify that the server latency is smaller than the client latency (which should always be true in real life anyway).

File: test/core/end2end/tests/filter_latency.c (hunk 3 above)
Comment (markdroth) [inline test/core/end2end/tests/filter_latency.c]: Done.
